game/cards/dm06/light_bringer.go

```diff
@@ -0,0 +1,102 @@
+package dm06
+
+import (
+	"duel-masters/game/civ"
+	"duel-masters/game/family"
+	"duel-masters/game/fx"
+	"duel-masters/game/match"
+)
+
+func AdomisTheOracle(c *match.Card) {
+
+	c.Name = "Adomis, the Oracle"
+	c.Power = 2000
+	c.Civ = civ.Light
+	c.Family = []string{family.LightBringer}
+	c.ManaCost = 3
+	c.ManaRequirement = []string{civ.Light}
+	c.TapAbility = true
+
+	c.Use(fx.Creature, fx.When(fx.TapAbility, func(card *match.Card, ctx *match.Context) {
+
+		shields := fx.SelectBackside(
+			card.Player,
+			ctx.Match,
+			card.Player,
+			match.SHIELDZONE,
+			"Adomis, the Oracle: Select one of your shields that will be shown to you",
+			1,
+			1,
+			true,
+		)
+
+		ids := make([]string, 0)
+
+		for _, s := range shields {
+			ids = append(ids, s.ImageID)
+		}
+
+		ctx.Match.ShowCards(
+			card.Player,
+			"Your shield:",
+			ids,
+		)
+
+		card.Tapped = true
+
+	}))
+}
+
+func VessTheOracle(c *match.Card) {
+
+	c.Name = "Vess, the Oracle"
+	c.Power = 2000
+	c.Civ = civ.Light
+	c.Family = []string{family.LightBringer}
+	c.ManaCost = 1
+	c.ManaRequirement = []string{civ.Light}
+
+	c.Use(fx.Creature, fx.Blocker, fx.CantAttackPlayers)
+}
+
+func YulukTheOracle(c *match.Card) {
+
+	c.Name = "Yuluk, the Oracle"
+	c.Power = 2500
+	c.Civ = civ.Light
+	c.Family = []string{family.LightBringer}
+	c.ManaCost = 1
+	c.ManaRequirement = []string{civ.Light}
+
+	spellcast := false
+
+	c.Use(fx.Creature, func(card *match.Card, ctx *match.Context) {
+
+		ctx.Match.ApplyPersistentEffect(func(ctx2 *match.Context, exit func()) {
```

Comment:
I don't think we need to use a persistent effect here, should be enough to listen to SpellCast and EndOfTurn event to set spellcast to true/false. Also, a new persistent effect is being added for every event that is being fired in the match here which is probably not what was intended